ee/desktop/runner/server/server.go

```diff
@@ -36,21 +35,21 @@ type controlRequestIntervalOverrider interface {
 	SetControlRequestIntervalOverride(time.Duration, time.Duration)
 }
 
-func New(logger log.Logger, controlRequestIntervalOverrider controlRequestIntervalOverrider) (*RunnerServer, error) {
+func New(slogger *slog.Logger, controlRequestIntervalOverrider controlRequestIntervalOverrider) (*RunnerServer, error) {
 	listener, err := net.Listen("tcp", "localhost:0")
 	if err != nil {
 		return nil, fmt.Errorf("creating net listener: %w", err)
 	}
 
 	rs := &RunnerServer{
 		listener:                        listener,
-		logger:                          logger,
+		slogger:                         slogger,
 		desktopProcAuthTokens:           make(map[string]string),
 		controlRequestIntervalOverrider: controlRequestIntervalOverrider,
 	}
 
-	if rs.logger != nil {
-		rs.logger = log.With(rs.logger, "component", "desktop_runner_root_server")
+	if rs.slogger != nil {
```

Comment:
Do we need to enforce that rs.slogger is not nil, or will calls to rs.slogger.Log below fail gracefully when nil instead of panicking?

Reply:
good catch, now returning error if slogger is nil